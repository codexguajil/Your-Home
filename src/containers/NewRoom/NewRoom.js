import React from 'react';
import { Button, Modal, Form, Input } from 'antd';
import { addNewRoom } from '../../thunks/addNewRoom';
import { connect } from 'react-redux';

const CollectionCreateForm = Form.create({ name: 'form_in_modal' })(
  class extends React.Component {
    render() {
      const { visible, onCancel, onCreate, form, newRoom } = this.props;
      const { getFieldDecorator } = form;
      return (
        <Modal
          visible={visible}
          title="Create a Room"
          okText="Create"
          onCancel={onCancel}
          onOk={onCreate}
        >
          <Form layout="vertical">
            <Form.Item label="Type">
              {getFieldDecorator('type', {
                rules: [{ required: true, message: 'Please input the type of room!' }],
              })(<Input onChange={newRoom} name='type'/>)}
            </Form.Item>
            <Form.Item label="Name">
              {getFieldDecorator('name', {
                rules: [{ required: true, message: 'Please input the name of the room!' }],
              })(<Input onChange={newRoom} name='name'/>)}
            </Form.Item>
            <Form.Item label="Description">
              {getFieldDecorator('description')(<Input onChange={newRoom} name='description' type="textarea" />)}
            </Form.Item>
            <Form.Item className="collection-create-form_last-form-item">
            </Form.Item>
          </Form>
        </Modal>
      );
    }
  }
);

export class NewRoom extends React.Component {
  state = { 
    visible: false,
    type: '',
    name: '',
    description: '' 
  };

  showModal = () => {
    this.setState({ visible: true });
  }

  newRoom = (e) => {
    this.setState({[e.target.name]: e.target.value});
  }

  handleCancel = () => {
    this.setState({ visible: false });
  }

  handleCreate = () => {
    const { name, type, description } = this.state;
    const { id } = this.props;
    this.props.addNewRoom(id, name, description, type);
    this.setState({ visible: false});
  }

  render() {
    return (
      <div>
        <Button type="primary" className='header-btn' onClick={this.showModal}>
          Create New Room
        </Button>
        <CollectionCreateForm
          visible={this.state.visible}
          onCancel={this.handleCancel}
          onCreate={this.handleCreate}
          newRoom={this.newRoom}
        />
      </div>
    );
  }
}

export const mapDispatchToProps = dispatch => ({
  addNewRoom: (id, name, description, type) => dispatch(addNewRoom(id, name, description, type))
});

export default connect(null, mapDispatchToProps)(NewRoom);