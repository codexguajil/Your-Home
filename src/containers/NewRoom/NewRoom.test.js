import React from 'react';
import { shallow } from 'enzyme';
import NewRoom from './NewRoom';

describe('NewRoom', () => {
  let wrapper;
  beforeEach(() => {
    wrapper = shallow(<NewRoom/>)
  });

  it('should match snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('showModal', () => {
    wrapper.instance().showModal();
    expect(wrapper.state('visible')).toBe(true);
  });

  it('handleCancel', () => {
    wrapper.instance().handleCancel();
    expect(wrapper.state('visible')).toBe(false);
  });

  it('saveFormRef', () => {

  });

  it('handleCreate', () => {

  });
  
});