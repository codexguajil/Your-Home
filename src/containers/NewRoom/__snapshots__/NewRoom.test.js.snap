// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NewRoom should match snapshot 1`] = `
<div>
  <Button
    block={false}
    className="header-btn"
    ghost={false}
    htmlType="button"
    loading={false}
    onClick={[Function]}
    type="primary"
  >
    Create New Room
  </Button>
  <Form(Component)
    newRoom={[Function]}
    onCancel={[Function]}
    onCreate={[Function]}
    visible={false}
  />
</div>
`;
